enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,630 @@
+---
+title: Cluster Logging Loki as an alternative Log Store
+authors:
+  - "@periklis"
+reviewers:
+  - "@ewolinetz"
+  - "@blockloop"
+  - "@lukas-vlcek"
+  - "@sichvoge"
+approvers:
+  - "@ewolinetz"
+  - "@alanconway"
+  - "@jcantril"
+creation-date: 2021-03-29
+last-updated: 2021-03-29
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Cluster Logging: Loki as an alternative Log Store
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of this is to provide an alternative low-retention log store based on [Loki](https://github.com/grafana/loki) for the cluster logging stack. This aims to provide a cloud-native first solution specialized in storing logs and in turn lessen the operational cost running cluster logging.
+
+Currently our main log store, i.e. Elasticsearch, is a more general-purpose ingestion, indexing and querying cluster-based solution for text search. It is a powerful solution with fine-grained control on how to scale log storage, indexing and querying.
+
+In general it serves its purpose quite well but it is a constant source of labor regarding operations. Its powerfulness and complexity requires a constant toll on the OpenShift cluster administrator side to understand how to individually scale the Elasticsearch cluster as a log store.
```

Comment:
We don't need to make the comparison encompassing all the angles. I selected the most representative issues that we have on production systems today. Yes, virtual/physical hardware cost is important but it will remain high for Loki, too. Only local disk space will get a huge relief, but the cost for object storage will eliminate this IMHO.

Reply:
@bwplotka To be really honest, when I saw first iteration of @periklis's HW proposal for Loki integration I thought to myself "If only we ever gave ES cluster so much resources (especially CPU...)". BTW did you know OCP default budget for single ES node was 1/2 CPU? (Why this was happening is a different story...) Going forward, if we will want to make any claims about how much Loki is doing better than ES we should really compare 1-1 on the resources level.